### PR TITLE
fix bug during download nuget

### PR DESCRIPTION
**Pull request by 928PJY**
$lastexitcode won't change by PowerShell command, if some error happens, an exception will be thrown and script stop.

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -65,8 +65,7 @@ if (-not(ValidateCommand($nugetCommand))) {
     mkdir -Path "$env:LOCALAPPDATA\Nuget" -Force
     $ProgressPreference = 'SilentlyContinue'
     [Net.WebRequest]::DefaultWebProxy.Credentials = [Net.CredentialCache]::DefaultCredentials
-    Invoke-WebRequest 'https://dist.nuget.org/win-x86-commandline/latest/nuget.exe' -OutFile $nugetCommand 
-    ProcessLastExitCode $lastexitcode "Download nuget.exe"
```
Comment (vicancy) [inline build.ps1]: what's the side effect check $lastexitcode?

Comment (928PJY) [inline build.ps1]: Invoke-WebRequest won't change the value of $lastexitcode even if some error happens
So if $lastexitcode default to be `not 0`, even if Invoke-WebRequest runs successfully,  an error message will be print and this script will exit.

---
In reply to: [127640655](https://github.com/dotnet/docfx/pull/1881#discussion_r127640655) [](ancestors = 127640655)

File: build.ps1 [hunk 2]
```diff
@@ -65,8 +65,7 @@ if (-not(ValidateCommand($nugetCommand))) {
...
+    Invoke-WebRequest 'https://dist.nuget.org/win-x86-commandline/latest/nuget.exe' -OutFile $nugetCommand
```
Comment (hellosnow) [inline build.ps1]: >Invoke [](start = 4, length = 6)

For the case such as uri is not avaiable, need we catch exception and update last exit code?

File: build.ps1 (hunk 1 above)
Comment (vicancy) [inline build.ps1]: so $lastexitcode is `not 0` from last command line call, we should exit with that?

**Comment by hellosnow** [comment]
:shipit:

File: build.ps1 (hunk 1 above)
Comment (hellosnow) [inline build.ps1]: >$lastexitcode [](start = 24, length = 13)

initialize $LastExitCode=0 in the very first beginning?

Comment (928PJY) [inline build.ps1]: this is the first time using `$lastexitcode`, so it depends on the default value.

---
In reply to: [127884787](https://github.com/dotnet/docfx/pull/1881#discussion_r127884787) [](ancestors = 127884787)

**Pull request merged**